lib/ruby_indexer/lib/ruby_indexer/configuration.rb

```diff
@@ -149,11 +110,23 @@ def indexable_uris
         end
       end
 
+      # These file names match our exclude regex, but often define important test parent classes that we need for
+      # ancestor linearization to provide test explorer functionality
+      excluded_from_ignore_test_files = ["test_case.rb", "test_helper.rb"]
+
       # Remove user specified patterns
+      bundle_path = Bundler.settings["path"]&.gsub(/[\\]+/, "/")
       uris.reject! do |indexable|
-        excluded_patterns.any? do |pattern|
-          File.fnmatch?(pattern, T.must(indexable.full_path), File::FNM_PATHNAME | File::FNM_EXTGLOB)
+        path = T.must(indexable.full_path)
+
+        # Don't exclude the special files, unless they are under fixtures or inside a gem installed inside of the
+        # workspace
+        if excluded_from_ignore_test_files.include?(File.basename(path)) &&
```

Comment:
This is a complex conditional, I think we can improve it by extracting out some helper methods or explaining variables.